### PR TITLE
swarm/network: update Kademlia EachBin and change start syncing

### PR DESCRIPTION
**Pull request by janos**

### CODE REVIEW

File: swarm/network/kademlia.go [hunk 1]
```diff
@@ -329,10 +329,39 @@ func (k *Kademlia) Off(p OverlayConn) {
 	}
 }
 
-func (k *Kademlia) EachBin(base []byte, pof pot.Pof, o int, eachBinFunc func(po, size int, f func(func(val pot.Val, i int) bool) bool) bool) {
+func (k *Kademlia) EachBin(base []byte, pof pot.Pof, o int, eachBinFunc func(conn OverlayConn, po int) bool) {
 	k.lock.RLock()
 	defer k.lock.RUnlock()
-	k.conns.EachBin(base, pof, o, eachBinFunc)
+
+	var i int
```
Comment (gbalint) [inline swarm/network/kademlia.go:336]: I think this is unused

File: swarm/network/stream/stream.go [hunk 2]
```diff
@@ -84,6 +83,15 @@ func NewRegistry(addr *network.BzzAddr, delivery *Delivery, db *storage.DBAPI, i
 	})
 	RegisterSwarmSyncerServer(streamer, db)
 	RegisterSwarmSyncerClient(streamer, db)
+
+	if doSync {
+		go func() {
+			time.Sleep(30 * time.Second)
```
Comment (gbalint) [inline swarm/network/stream/stream.go]: If this is a temporary hack let's add a comment about it

File: swarm/network/kademlia.go (hunk 1 above)
Comment (gbalint) [reply swarm/network/kademlia.go:336]: No, it is actually used, sorry

File: swarm/network/stream/stream.go [hunk 3]
```diff
@@ -84,6 +83,15 @@ func NewRegistry(addr *network.BzzAddr, delivery *Delivery, db *storage.DBAPI, i
 	})
 	RegisterSwarmSyncerServer(streamer, db)
 	RegisterSwarmSyncerClient(streamer, db)
```
Comment (holisticode) [inline swarm/network/stream/stream.go:85]: Here we do `RegisterSwarmSyncerServer` and Client - do we actually also need to do it in `common_test.go` line 88?

Comment (janos) [reply swarm/network/stream/stream.go:85]: Yes, you are right. Creating a new PR.

**Pull request merged**